src/commands.rs

```diff
@@ -1615,44 +1616,64 @@ struct BackoutArgs {
     destination: Vec<String>,
 }
 
-/// Create, update, or delete a branch
+/// Manage branches.
 ///
 /// For information about branches, see
 /// https://github.com/martinvonz/jj/blob/main/docs/branches.md.
-#[derive(clap::Args, Clone, Debug)]
-struct BranchArgs {
-    /// The branch's target revision
-    #[clap(long, short, group = "action")]
-    revision: Option<String>,
-
-    /// Allow moving the branch backwards or sideways
-    #[clap(long, requires = "revision")]
-    allow_backwards: bool,
-
-    /// Delete the branch locally
+#[derive(clap::Subcommand, Clone, Debug)]
+enum BranchSubcommand {
+    /// Create a new branch.
+    #[clap(visible_alias("c"))]
+    Create {
+        /// The branch's target revision.
+        #[clap(long, short)]
+        revision: Option<String>,
+
+        /// The branches to create.
+        names: Vec<String>,
+    },
+
+    /// Delete an existing branch and propagate the deletion to remotes on the next push.
+    #[clap(visible_alias("d"))]
+    Delete {
+        /// The branches to delete.
+        names: Vec<String>,
+    },
+
+    /// Delete the local version of an existing branch, without propagating the
+    /// deletion to remotes.
```

Comment:
It looks like the description is wrapped for `forget`. Do you want to avoid that?

```
$ cargo run -- branch -h       
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/jj branch -h`
jj-branch 
Manage branches

USAGE:
    jj branch [OPTIONS] <SUBCOMMAND>

GLOBAL OPTIONS:
        --at-operation <AT_OPERATION>    Operation to load the repo at [default: @]
    -h, --help                           Print help information, more help with --help than with -h
        --no-commit-working-copy         Don't commit the working copy
    -R, --repository <REPOSITORY>        Path to repository to operate on

SUBCOMMANDS:
    create    Create a new branch [aliases: c]
    delete    Delete an existing branch and propagate the deletion to remotes on the next push
                  [aliases: d]
    forget    Delete the local version of an existing branch, without propagating the deletion
                  to remotes [aliases: f]
    help      Print this message or the help of the given subcommand(s)
    list      List branches and their targets [aliases: l]
    set       Update a given branch to point to a certain commit [aliases: s]
```

Reply:
I'm not sure. I looked at `git help` and `hg help` and a few others I looked at all seem to try to keep it to one line, but I'm not sure if there's a good reason for that :) I'm fine with either.

Reply:
> but I'm not sure if there's a good reason for that :)

And I do think there's good reason to let these be longer and clearer. So I guess I'd suggest leaving them as you had them, unless you feel that we should mimic the other tools.